src/base/math/grnd.h

```diff
@@ -38,22 +39,22 @@ template<> __host__ __device__ inline double minVal<double>(){ return DBL_MIN; }
 /**
 * internal functions, should only be called from get_rand!
 */
-__device__ __host__ inline unsigned taus_step( unsigned &z, int S1, int S2, int S3, unsigned M)
+__host__ __device__ inline unsigned taus_step( unsigned &z, int S1, int S2, int S3, unsigned M)
```

Comment:
Why did you swap __device__ <-> __host__ everywhere?

Reply:
This is a byproduct of my previous define replacements

Reply:
Ok, I've reverted these swaps. Otherwise it's very hard to find the relevant changes for me